codec/dagcbor/roundtripCidlink_test.go

```diff
@@ -15,27 +14,26 @@ import (
 )
 
 func TestRoundtripCidlink(t *testing.T) {
-	lb := cidlink.LinkBuilder{cid.Prefix{
+	lp := cidlink.LinkPrototype{cid.Prefix{
 		Version:  1,
 		Codec:    0x71,
 		MhType:   0x17,
 		MhLength: 4,
```

Comment:
can this not be inferred from the type?

Reply:
`{` not `(` :) it's an adapter type, not an interface cast.

Reply:
i think i was wanting the apparently deprecated '-1' to indicate default length because a lot of these hashes have a single length that makes sense, and i don't want to be remembering / fumble that sha1 should be 20 bytes while sha224 should be 28 bytes every time i make one of these.

Reply:
oh, sorry, I thought the comment was on the diff line, since that's what github highlighted most brightly.

Yeah, agree.  I wish the go-cid and/or go-multihash libraries were more friendly about this very common user story.

I think a -1 should flow through and do whatever go-multihash does, still.  And I have no idea why that's deprecated, fwiw.  (A lot of things in go-multihash seem deprecated without much comment on why or what to do instead.  I think some review and renovation of that is overdue.)

It's slightly on the other side of where I'm cordoning my renovation today, though.